### PR TITLE
Bootstrap4 improvements

### PR DESCRIPTION
**Pull request by ddworak**

### CODE REVIEW

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala [hunk 1]
```diff
@@ -83,29 +91,29 @@ abstract class TooltipUtils[TooltipType <: Tooltip[_, TooltipType]] {
     */
   def apply(
     animation: Boolean = true,
-    boundary: String = "scrollParent",
-    container: Option[String] = None,
-    content: dom.Node => String | dom.Node = _ => "",
+    boundary: String | dom.Node = "scrollParent",
+    container: Option[String | dom.Node] = None,
+    content: js.Function1[dom.Node, String] | dom.Node = io.udash.emptyStringNode(),
     delay: Delay = Delay(0 millis, 0 millis),
```
Comment (Kurtzz) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala]: `delay` can be just a number too

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala [hunk 2]
```diff
@@ -83,29 +91,29 @@ abstract class TooltipUtils[TooltipType <: Tooltip[_, TooltipType]] {
...
     html: Boolean = false,
     offset: String = "0",
```
Comment (Kurtzz) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala]: `offset` can be a number too

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/BootstrapStyles.scala [hunk 3]
```diff
@@ -672,3 +597,94 @@ object BootstrapStyles {
...
+    final val Info: Value = new Color("-info")
+    final val Light: Value = new Color("-light")
+    final val Dark: Value = new Color("-dark")
+    final val White: Value = new Color("-white")
+    /** This value is only supported in buttons. */
+    final val Link: Value = new Color("-link")
+  }
+
+  final class Size(val classMarker: String)(implicit enumCtx: EnumCtx) extends AbstractValueEnum
+  object Size extends AbstractValueEnumCompanion[Size] {
```
Comment (Kurtzz) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/BootstrapStyles.scala]: `xl` missing
Modal accepts xl size: [Modal](https://getbootstrap.com/docs/4.3/components/modal/#optional-sizes)

**Comment by Kurtzz** [comment]
### Alert
- `alert-heading` class missing

### Button
- `btn-group-toggle` classmissing ([here](https://getbootstrap.com/docs/4.3/components/buttons/#checkbox-and-radio-buttons))

- toogle can be achive by adding `data-toggle="button" aria-pressed="false"` instead of manually toggle `active'`attribute ([here](https://getbootstrap.com/docs/4.3/components/buttons/#toggle-states))

### Dropdown

- `dropdown-item-text` class missing
- `btn-group` in top level div of the Component is useless
- whether `<button>` or `<a>` shouldn't depends on `dataToggle` property
`data-toggle` is just a css class which adds an arrow to a button. Consider something like a `ButtonTag` 
- there are only methods with `items` as constructor. What about dropdown with some div with [form](https://getbootstrap.com/docs/4.3/components/dropdowns/#form), period picker or [something](https://getbootstrap.com/docs/4.3/components/dropdowns/#text)? 

### Borders

- `rounded-pill` class missing
- rounded size classes missing (`rounded-sm`, `rounded-lg`)

### Shadows
- all classes missing [here](https://getbootstrap.com/docs/4.3/utilities/shadows/)

### Text
- `text-wrap`, `text-break`, `text-reset` are missing

### Typography
- `display-{}` (1,2,3,4) missing [here](https://getbootstrap.com/docs/4.3/content/typography/#display-headings)

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/BootstrapStyles.scala (hunk 3 above)
Comment (ddworak) [reply bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/BootstrapStyles.scala]: It doesn't on Boostrap 4.1, migration will be another thing.

**Comment by ddworak** [comment]
Added:
- `alert-heading`
- `btn-group-toggle`
- Shadows
- `display-{}`
- `dropdown-item-text` (also to default item factory)
- `Raw(element` dropdown item

Removed:
- `btn-group` in top level dropdown div

I'm not sure about button toggling with `data-toggle` - I'd rather stick with current approach as it's simple to align with the `active` property.

Unavailable in 4.1:
- `rounded-pill`
- `rounded-sm`, `rounded-lg`
- `text-wrap`
- `text-break`
- `text-reset`
- `-xl` for modals

**Comment by Kurtzz** [comment]
> Unavailable in 4.1:
> 
> * `rounded-pill`
> * `rounded-sm`, `rounded-lg`
> * `text-wrap`
> * `text-break`
> * `text-reset`
> * `-xl` for modals

Can we upgrade Bootstrap version to 4.3.0 therefore?

**Comment by ddworak** [comment]
@Kurtzz there's some unfinished business with 4.2 https://github.com/UdashFramework/udash-core/pull/252, I guess we'll go all the way to whatever's latest once we get to it

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/BootstrapStyles.scala [hunk 4]
```diff
@@ -638,6 +566,11 @@ object BootstrapStyles {
   }
 
   object Typography {
+    def display1 = CssStyleName("display1")
```
Comment (Kurtzz) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/BootstrapStyles.scala]: Should be `display-1` with a dash etc.

**Pull request merged**